core/lib/scmi-server/sub.mk

```diff
@@ -0,0 +1,174 @@
+# SCMI server library is built from SCP-firmware source tree.
+# The firmware is made of the framework, a product and modules.
+# Only module used by firmware must be built, see _CFG_SCPFW_MOD_*.
+
+scpfw-path = $(CFG_SCP_FIRMWARE)
+scpfw-product = $(CFG_SCMI_SCPFW_PRODUCT)
+scpfw-out-path := $(out-dir)/$(libdir)
+
+# This script was validated against SCP-firmware 2.11.0 development branch,
+# from commit f1d894921d76 ("product/optee-fvp: Add new OPTEE FVP product").
+scpfw-integ-version-maj = 2
+scpfw-integ-version-min = 11
+scpfw-integ-version-pat = 0
+scpfw-integ-version = $(scpfw-integ-version-maj).$(scpfw-integ-version-min).$(scpfw-integ-version-pat)
+
+srcs-y += scmi_server.c
+incdirs-y += include
+incdirs_ext-y += $(scpfw-path)/arch/none/optee/include
+
+# SCP-firmware cmake configuration generates header fwk_module_idx.h and
+# source files fwk_module_list.c needed for scp-firmware compilation.
+scpfw-cmake-flags-y = -DSCP_FIRMWARE_SOURCE_DIR:PATH=$(scpfw-product)/fw \
+		      -DSCP_LOG_LEVEL="TRACE" \
+		      -DDISABLE_CPPCHECK=1 \
+		      -DCFG_NUM_THREADS=$(CFG_NUM_THREADS) \
+		      -DSCP_OPTEE_DIR:PATH=$(CURDIR) \
+		      -DCFG_CROSS_COMPILE=$(lastword $(CROSS_COMPILE_core))
+
+gensrcs-y += fwk_module_list
+force-gensrc-fwk_module_list := y
+produce-fwk_module_list = build/framework/src/fwk_module_list.c
+depends-fwk_module_list = FORCE
+recipe-fwk_module_list = cmake -S $(scpfw-path) -B $(scpfw-out-path)/build $(scpfw-cmake-flags-y)
+# Include Generated header file fwk_module_idx.h
+incdirs_ext-y += $(scpfw-out-path)/build/framework/include
+
+cflags-lib-y += -DBUILD_VERSION_MAJOR=$(scpfw-integ-version-maj) \
+		-DBUILD_VERSION_MINOR=$(scpfw-integ-version-min) \
+		-DBUILD_VERSION_PATCH=$(scpfw-integ-version-pat)
+
+scpfw-impl-version := $(shell git -C $(scpfw-path) describe --always --dirty=-dev 2>/dev/null || \
+                      echo Unknown_$(scpfw-integ-version))
+cflags-lib-y += -DBUILD_VERSION_DESCRIBE_STRING=\"$(scpfw-impl-version)\"
+
+cflags-lib-y += -DFWK_LOG_LEVEL=$(_CFG_SCPFW_LOG_LEVEL)
+
+cflags-lib-y += -Wno-cast-align \
+		-Wno-nonnull-compare \
+		-Wno-unused-parameter \
+		-Wno-missing-prototypes \
+		-Wno-missing-declarations \
+		-Wno-suggest-attribute=format \
+		-Wno-declaration-after-statement
+
+# The below directives will be removed once SCP-firmware pull requests
+# 728, 729 and 730 are merged.
+cflags-lib-y += -Wno-undef \
+		-Wno-maybe-uninitialized \
+		-Wno-unused-but-set-variable \
+		-Wno-suggest-attribute=format \
+
+cflags-lib-$(_CFG_SCPFW_NOTIFICATION) += -DBUILD_HAS_NOTIFICATION \
```

Comment:
This should go into `cppflags-lib-...`, same below.

Reply:
ok, thanks.